vyxal/elements.py

```diff
@@ -5338,199 +5335,50 @@ def vectorise(
-                for x in lhs
-            ),
-            (SCALAR_TYPE, SCALAR_TYPE, list, list): lambda: (
-                safe_apply(function, lhs, rhs, x, y, ctx=ctx)
-                for x, y in vy_zip(other, fourth, ctx=ctx)
-            ),
-            (SCALAR_TYPE, list, SCALAR_TYPE, list): lambda: (
-                safe_apply(function, lhs, x, other, y, ctx=ctx)
-                for x, y in vy_zip(rhs, fourth, ctx=ctx)
-            ),
-            (list, SCALAR_TYPE, SCALAR_TYPE, list): lambda: (
-                safe_apply(function, x, rhs, other, y, ctx=ctx)
-                for x, y in vy_zip(lhs, fourth, ctx=ctx)
-            ),
-            (SCALAR_TYPE, list, list, SCALAR_TYPE): lambda: (
-                safe_apply(function, lhs, x, y, fourth, ctx=ctx)
-                for x, y in vy_zip(rhs, other, ctx=ctx)
-            ),
-            (list, SCALAR_TYPE, list, SCALAR_TYPE): lambda: (
-                safe_apply(function, x, rhs, y, fourth, ctx=ctx)
-                for x, y in vy_zip(lhs, fourth, ctx=ctx)
-            ),
-            (list, list, SCALAR_TYPE, SCALAR_TYPE): lambda: (
-                safe_apply(function, x, y, other, fourth, ctx=ctx)
-                for x, y in vy_zip(lhs, rhs, ctx=ctx)
-            ),
-            (list, list, list, SCALAR_TYPE): lambda: (
-                safe_apply(function, x, y, z, fourth, ctx=ctx)
-                for x, y, z in transpose([lhs, rhs, other], ctx=ctx)
-            ),
-            (list, list, SCALAR_TYPE, list): lambda: (
-                safe_apply(function, x, y, other, z, ctx=ctx)
-                for x, y, z in transpose([lhs, rhs, fourth], ctx=ctx)
-            ),
-            (list, SCALAR_TYPE, list, list): lambda: (
-                safe_apply(function, x, rhs, y, z, ctx=ctx)
-                for x, y, z in transpose([lhs, other, fourth], ctx=ctx)
-            ),
-            (SCALAR_TYPE, list, list, list): lambda: (
-                safe_apply(function, lhs, x, y, z, ctx=ctx)
-                for x, y, z in transpose([rhs, other, fourth], ctx=ctx)
-            ),
-            (list, list, list, list): lambda: (
-                (
-                    safe_apply(function, w, x, y, z, ctx=ctx)
-                    for (w, x), (y, z) in vy_zip(
-                        vy_zip(lhs, rhs, ctx=ctx),
-                        vy_zip(other, fourth, ctx=ctx),
-                    )
-                )
-                if ctx.double_zip_vectorize
-                else (
-                    safe_apply(function, w, x, y, z, ctx=ctx)
-                    for w, x, y, z in transpose(
-                        [lhs, rhs, other, fourth], ctx=ctx
-                    )
-                )
-            ),
-        }
-
-        if explicit:
+    if explicit:
+        # explicit vectorisation
+        # Algorithm: If there are no lists, vectorise over lhs.
+        # Else, vectorise over the first list.
+        ts = [primitive_type(x) for x in args]
```

Comment:
Rip vectorisation over infinite lists

Reply:
Why? This is iterating over each argument to vectorise. Are you referring to another line?

Reply:
Oh.


I thought it was going over items in the list that's passed in